### ISSUE TITLE
Feature: heredoc

### ISSUE THREAD

**Issue by aequitas** [purpleidea/mgmt#364]
Lang should support some kind of heredoc, eg:

```
$x = template(@(EOF)
    some text
    that spans mulitple lines
    but with leading whitespace removed
    |EOF
)
```

Reference: https://github.com/puppetlabs/puppet-specifications/blob/master/language/heredoc.md

**Comment by purpleidea** [comment]
@aequitas I was thinking about it, but don't know how to do this in the lexer. We'd have to agree on syntax first, and not copy puppet.

**Comment by aequitas** [comment]
Agree, first syntax/design needs to be discussed. I think some good parts can be taken from the current Puppet syntax, but not everything might be needed. For MVP leading whitespace stripping should be a feature imho.

**Comment by purpleidea** [comment]
I find that avoiding thinking about what others does is a good way to think about design. Come up with a novel approach, and THEN compare after if necessary. Don't get tunnel visioned into something you already know.

Everyone agrees about tab prefix stripping :)
